### DISCUSSION TITLE
Studying Linux Mint Cinnamon Spices frontend-development/design workflows

### DISCUSSION THREAD

**Discussion by AltCaviar** [linuxmint/discussions#368] [Development]
I am a student studying Linux Mint Cinnamon Spices frontend-development workflows. To understand desktop frontend-development workflow standardisation. To broaden inclusion in open-source graphical user interface design and development.

I have looked in 'Devhelp' and could not find anything, This raised questions about workflows implemented in their development. I was told on Linux Mint Forums my questions would be more appropriate to post on Linux Mint: Discussions. about the front end development of themes. 

1. Is there a standardised workflow by which Themes on the Cinnamon Spices are developed (visual concept to development)?
2. Is there documentation or a list of required design elements such as icons, menus, buttons, window-boarders, and so forth to be considered by developers when approaching a new Theme?
3. Are there comprehensive templates that front end development can use in design briefing?
4. What documentation are front end developers providing to back-end developers in the creation of Themes?
5. If this documentation does exist, could I see them?
6. If they do not exist, would I be beneficial for me to develop it?

Thank you for your time in responding this question, I hope I have posed it in the right place this time. I Look forward to hearing back from you.

**Comment by JosephMcc** [comment]
> Is there a standardised workflow by which Themes on the Cinnamon Spices are developed (visual concept to development)?

No. The best way is to start with the default theme or a theme that's close to what you want and start modifying it.

> Is there documentation or a list of required design elements such as icons, menus, buttons, window-boarders, and so forth to be considered by developers when approaching a new Theme?

I think you are talking about different things. Those are supplied by different themes. Icons by icon themes, window borders by Metacity or xfwm themes (not part of Cinnamon), the desktop which it the Cinnamon theme, and the windows which are the Gtk theme. The spices site was originally meant just for sharing Cinnamon themes.

> Are there comprehensive templates that front end development can use in design briefing?

Not that I know of

> What documentation are front end developers providing to back-end developers in the creation of Themes?

For Cinnamon themes, none really. They aren't that complicated.

> If they do not exist, would I be beneficial for me to develop it?

Personally not sure it would really be worth the effort. Cinnamon themes aren't that complicated and the documentation probably wouldn't get updated/maintained very well when new things are added.
